dask/array/core.py

```diff
@@ -165,6 +167,27 @@ def decorator(dask_func):
     return decorator
 
 
+def check_if_handled_given_other(f):
+    """ Check if method is handled by Dask given type of other
+
+    Ensures proper deferal to upcast types in dunder operations without
+    assuming unknown types are automatically downcast types.
+    """
+
+    @wraps(f)
+    def wrapper(self, other):
+        if (
+            is_valid_array_chunk(other)
+            or isinstance(other, (self.__class__, list, tuple) + np.ScalarType)
```

Comment:
`np.ScalarType` may miss custom `dtype`s in the future.

```suggestion
            or isinstance(other, (self.__class__, list, tuple, np.generic))
```